apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/LocationHelperModule.kt

```diff
@@ -0,0 +1,267 @@
+
+/**
+ * Android Native module that provides:
+ * - a method to check and enable if necessary the location service
+ * - a method to listen to the state (enabled/disabled) of the location service
+ *
+ * See from https://developers.google.com/android/reference/com/google/android/gms/location/package-summary :
+ * - SettingsClient which provides a way to interact with the location settings-enabler APIs
+ * - FusedLocationProviderClient which provides a way to listen to the location service state updates
+ */
+class LocationHelperModule(private val reactContext: ReactApplicationContext) :
+    ReactContextBaseJavaModule(reactContext) {
+
+    // Resolves with a success or error value. reject is not used because we want to pass an
+    // information to the JS code.
+    private var locationPromise: Promise? = null
+
+    // LocationRequest used by both LocationSettings and FusedLocationProviderClient
+    // LocationRequest.create() is deprecated, but LocationRequest.Builder()
+    // is available from `import android.location.LocationRequest` which
+    // requires API level >= 31 (current min is 24)
+    private var locationRequest: LocationRequest = LocationRequest.create();
+
+    private var fusedLocationProviderClient: FusedLocationProviderClient;
+
+    /**
+     * Listens to result from the location enable request activity.
+     * Request sent from requestEnableLocation.
+     */
+    private val activityEventListener = object : BaseActivityEventListener() {
+        override fun onActivityResult(
+            activity: Activity?,
+            requestCode: Int,
+            resultCode: Int,
+            data: Intent?
+        ) {
+            if (requestCode === REQUEST_ENABLE_LOCATION) {
+                when (resultCode) {
+                    Activity.RESULT_OK -> {
+                        locationPromise?.resolve(SUCCESS_LOCATION_ENABLED);
+                    }
+                    Activity.RESULT_CANCELED -> {
+                        locationPromise?.resolve(ERROR_USER_DENIED_LOCATION);
+                    }
+                }
+                locationPromise = null;
+                return
+            }
+            super.onActivityResult(activity, requestCode, resultCode, data)
+        }
+    }
+
+    /**
+     * Handler to listen to location services updates (turned on/off)
+     *
+     * Emits an event to JS everytime the state of the location services change
+     */
+    private val locationServicesUpdateCallback = object : LocationCallback() {
+        override fun onLocationAvailability(p0: LocationAvailability) {
+            val isLocationAvailable = p0.isLocationAvailable
+
+            if (isLocationAvailable) {
+                reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                    .emit("LocationServiceUpdated", Arguments.createMap().apply {
+                        putString("service", "enabled")
```

Comment:
why not just use a boolean here?

Reply:
I think it's more declarative than just saying "true/false", but also this value is serialised (IMO), so i don't think it changes anything